goodpeer/goodpeer.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python3
+from collections import OrderedDict
+from pyln.client import Plugin, Millisatoshi
+
+plugin = Plugin()
+
+
+def get_channel_infos(chan):
+    """
+    Filter out what we need from a channel entry in listchannels
+    """
+    return {
+        "scid": chan["short_channel_id"],
+        "ppm_fee": chan["fee_per_millionth"],
+        "base_fee": chan["base_fee_millisatoshi"],
+        "value": chan["amount_msat"],
+    }
+
+
+def get_nodes(plugin):
+    """
+    Get public Lightning network nodes.
+
+    :return: A dict with the node id as key, and a list of channels as values.
+    """
+    peers = {}
+    channels = plugin.rpc.listchannels()
+    for chan in channels["channels"]:
+        if chan["fee_per_millionth"] == chan["base_fee_millisatoshi"] == 0:
+            # They are likely an agreement or the same person, they'd just
+            # distort the medians.
+            continue
+        if not chan["active"]:
+            # It could just be temporary, but we don't want zombies!
+            continue
+        id = chan["source"]
+        if id not in peers.keys():
+            peers[id] = [get_channel_infos(chan)]
+        else:
+            peers[id].append(get_channel_infos(chan))
```

Comment:
This can usually be simplified to this:

```python
if id not in peers:
    peers[id] = []
peers[id].append(get_channel_infos(chan))
```

Or even more concise:

```python
peers[id] = peers.get(id, []).append(get_channel_infos(chan))
```

though the second one is a bit less legible imho :wink:

Reply:
Thanks!